epyqlib/devicetreeview.py

```diff
@@ -221,6 +230,28 @@ def add_device(self, bus, device=None):
 
         return device
 
+    def change_device_number(self, node):
+        pcan_bus = can.interface.Bus(bustype=node.interface, channel=node.channel)
+        current_device_number = pcan_bus.get_device_number()
+        new_device_number, ok = QInputDialog.getInt(
+            None,
+            *(("Device ID",) * 2),
+            current_device_number,
+            0,
+            255,
+        )
```

Comment:
I always forget the nuances of `exec`ing dialogs.  Does it continue handling all the outer activities including qt5reactor/twisted stuff?  Even if this bus is offline, others may be online and active.  Did Matt give you two PCANs?  `:]`  If not, we can manage another way to check.

Reply:
I can see your point. I only have one PCAN adapter. My guess is that it would not interfere because the "Converter Node ID" input dialog displays after the user selects a device file in a very similar fashion. I could test with a second adapter (tomorrow). What is your alternate way to test?